analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CloudNative/AzureFunctionsLogFailures.cs

```diff
@@ -0,0 +1,154 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace AzureFunctions1
+{
+    public static class Function1
+    {
+        [FunctionName("Function1")]
+        public static async Task<IActionResult> EmptyCatchClause([HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)] HttpRequest req, ILogger log)
+        {
+            try
+            {
+                return new EmptyResult();
+            }
+            catch // Noncompliant {{Log caught exceptions via ILogger with LogLevel Warning, Error, or Critical}}
+//          ^^^^^
+            {
+                return new EmptyResult();
+            }
+        }
+
+        [FunctionName("Function1")]
```

Comment:
Will it raise if I create `public interface ICustom : ILogger {}` and use it as a parameter? Let's add UT

Reply:
I don't think this would work in Azure Functions, as the ILogger gets passed from the infrastructure. A custom interface will likely be just `null` at runtime, or the function is not be called at all.

Reply:
We should still document the behavior in UT